### DISCUSSION TITLE
How to attach agent to AWS Lambda Function

### DISCUSSION THREAD

**Discussion by diegoquirino** [inspectIT/inspectit-ocelot#1216] [Q&A] [answered]
Hi everyone,

I have an AWS Lambda Function (that is a Spring Boot application) "hosted" into AWS Cloud using combination of API Gateway + Lambda to enable an API.

**How is it possible to attach inspectit "agent.jar" while deploying my lambda function to AWS?**

I'd tried to use template.yml section to set "-javaagent" but "*.jar" file never is found.

Thanks for help.

**Comment by SaschaBrechmannVHV** [comment, marked as answer]
hi @diegoquirino ,

we have the same problem and only a workaround.
There are some Problems with the inspect-IT Agent and AWS Lambda

1.  The agent need some time to start the instrumentation. So in our UseCase we cound not use the inspect-IT agent. We have a batchjob like AWS Lambda Function
2. The Agent needs some config. This could not be providet with the inspect-IT configsrv, only with file or environment.
3. The Trace-/Metrics-Receiver needs to be public available for AWS Lambda. This is no option for us.

So we decided to use the opentelemetry-Framework directly and not the Agent. (OK, the agent use the opencensus-framework. But a update to opentelemetry is on the roadmap.)

There are some good Docs from AWS how to include opentelemetry in AWS Lambda und how to send the Traces to an Receiver of our choise.

Regards, Sascha

**Comment by mariusoe** [comment]
Hi,

as @SaschaBrechmannVHV described, I would not recommend using the agent in lambda functions because of the time it requires to initialize and load the configuration. It is easier and less overhead to directly use the OpenTelemetry framework during implementation. But then you have to add the monitoring code into your codebase.

In case you want to use the agent, make sure that the agent JAR is available to the process of the lambda function. Then you should be able to add it via the javaagent property just like to a local JVM.
As far as I know, there are many example with other Java agents explaining how to add an agent to a Java lambda function.
